### PR TITLE
Added support for serializing ```NaN``` and ```Infinity``` to Json

### LINKED ISSUE #519: [RFC]: add support for serializing `NaN` and infinities to JSON
### Description

This RFC proposes add support for serializing `NaN` and infinities to JSON.

Currently, when serializing an object containing a `NaN`, the resultant value is `null`.

```javascript
In [1]: JSON.stringify( { 'x': NaN } )
Out[1]: '{"x":null}'
```

Similarly, when serializing an object containing either `+infinity` or `-infinity`, the resultant value is `null` (as documented in #520).

This is problematic when attempting to roundtrip an object data structure containing `NaN` and/or infinity values.

This RFC seeks to remedy this problem by standardizing how `NaN` and infinity values are serialized to JSON. This proposal builds on 

-    [`@stdlib/error/to-json`](https://github.com/stdlib-js/stdlib/tree/83aae89c6320379c5798d564761c5479854fb727/lib/node_modules/%40stdlib/error/to-json)
-   [`@stdlib/buffer/to-json`](https://github.com/stdlib-js/stdlib/tree/83aae89c6320379c5798d564761c5479854fb727/lib/node_modules/%40stdlib/buffer/to-json)

and proposes to serialize a `NaN` value as follows:

```text
{
    'type': 'float64',
    'value': 'NaN'
}
```

and infinity values as follows:

```
{
    'type': 'float64',
    'value': '+Infinity'
}
{
    'type': 'float64',
    'value': '-Infinity'
}
```

A corresponding JSON reviver (e.g., similar to [`@stdlib/buffer/reviver`](https://github.com/stdlib-js/stdlib/tree/83aae89c6320379c5798d564761c5479854fb727/lib/node_modules/%40stdlib/buffer/reviver)) can then recognize the standardization format and reconstitute a `NaN` value.

Package: `@stdlib/number/float64/to-json`
Alias: `num2json`

### Related Issues

No.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #519 .

## Description

With this PR, I am able to convert NaN, +Infinity, -Infinity to the json asked in #519 

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

@kgryte you can have a review :)

### CODE REVIEW

File: lib/node_modules/@stdlib/number/float64/to-json/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+'use strict';
+
+/**
+* Return a JSON representation of a `Buffer`.
+*
+* @module @stdlib/number/float64/to-json
+*
+* @example
+* var number2json = require( '@stdlib/number/float64/to-json' );
+* var str = JSON.stringify( number2json( NaN ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/index.js:29]:
```suggestion
* 
* var str = JSON.stringify( number2json( NaN ) );
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/index.js]:
```suggestion
* Copyright (c) 2022 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Return a JSON representation of a `Buffer`.
+*
+* @module @stdlib/number/float64/to-json
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/index.js:24]:
```suggestion
* @module @stdlib/number/to-json
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+* @throws {TypeError} type of x must be a number primitive
+* @returns {Object} JSON representation
+*
+* @example
+* var json = number2json( NaN );
+* // returns { 'type': 'float64', 'value': 'NaN' }
+*/
+function number2json( x ) {
+	if ( typeof x !== 'number' ) {
+		throw new TypeError( 'invalid argument. Must provide a number. Value: `' + x + '`.' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
		throw new TypeError( format( 'invalid argument. Must provide a number. Value: `%d`.', x ) );
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js:28]:
```suggestion
// MODULES //

var PINF = require( '@stdlib/constants/float64/pinf' );
var NINF = require( '@stdlib/constants/float64/ninf' );
var format = require( '@stdlib/string/format' );


// MAIN //
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+* Returns a JSON representation of a `Number`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
* Returns a JSON representation of a number.
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+*
+* @param {Number} x - number to serialize
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+* @throws {TypeError} type of x must be a number primitive
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
* @throws {TypeError} must provide a number
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+*
+* @example
+* var json = number2json( NaN );
+* // returns { 'type': 'float64', 'value': 'NaN' }
+*/
+function number2json( x ) {
+	if ( typeof x !== 'number' ) {
+		throw new TypeError( 'invalid argument. Must provide a number. Value: `' + x + '`.' );
+	}
+	if ( x!==x ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
	if ( x !== x ) {
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+		return {
+			'type': 'float64',
+			'value': 'NaN'
+		};
+	}
+	if ( x === Number.POSITIVE_INFINITY ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
	if ( x === PINF ) {
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+		return {
+			'type': 'float64',
+			'value': '+Infinity'
+		};
+	}
+	if ( x === Number.NEGATIVE_INFINITY ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
	if ( x === NINF ) {
```

**Comment by Pranavchiku** [comment]
So, I created ```repl.txt``` in docs and then when I commit it shows the following log, if you can help
```
(base) pranavchiku@pranavchiku-Yoga-Slim-7-14IIL05:~/stdlib$ git commit -m "created docs and added repl.txt"
node:internal/modules/cjs/loader:958
  throw err;
  ^

Error: Cannot find module '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/number/float64/to-json'
Require stack:
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/rules/doctest/lib/main.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/rules/doctest/lib/index.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/register_validators.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/linter.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/main.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/index.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/lint.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/async.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/index.js
- /home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/bin/cli
    at Module._resolveFilename (node:internal/modules/cjs/loader:955:15)
    at Module._load (node:internal/modules/cjs/loader:803:27)
    at Module.require (node:internal/modules/cjs/loader:1021:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.returnValues [as validate] (/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/rules/doctest/lib/main.js:120:17)
    at Linter.validate (/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/validate.js:110:32)
    at isValid (/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/main.js:42:9)
    at onRead (/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/lint.js:93:11)
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/rules/doctest/lib/main.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/rules/doctest/lib/index.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/register_validators.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/linter.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/main.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/repl-txt/validate/lib/index.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/lint.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/async.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/lib/index.js',
    '/home/pranavchiku/stdlib/lib/node_modules/@stdlib/_tools/lint/repl-txt/bin/cli'
  ]
}

Node.js v18.8.0

REPL help lint errors.
```

**Comment by Pranavchiku** [comment]
@kgryte right now, I am facing trouble in committing ```repl.txt``` for which the logs are shown above, I will now be adding ```benchmark``` and ```readme.md``` as soon as possible :)

File: lib/node_modules/@stdlib/number/float64/examples/index.js [hunk 12]
```diff
@@ -19,6 +19,16 @@
 'use strict';
 
 var objectKeys = require( '@stdlib/utils/keys' );
+var number2json = require( './../to-json/lib' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/examples/index.js]: You can revert these changes. Package examples should be place in the `examples/index.js` file local to the package, not in the namespace examples files.

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/// <reference types="node"/>
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts]:
```suggestion
```

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+/**
+* Returns a JSON representation of a number.
+*
+* @param x - input number to serialize
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts]:
```suggestion
* @param x - input value
```

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+* @returns JSON representation
+*
+* @example
+* var number2json = require( '@stdlib/number/float64/to-json' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts]:
```suggestion
```

In TypeScript declarations, we don't need to explicitly import the package. This is implied.

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/test.ts [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+// TESTS //
+
+// The function returns an object...
+{
+	const num = 3.14;
+	number2json( num ); // $ExpectType JSON
+}
+
+// The compiler throws an error if the function is provided a value other than a buffer...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/test.ts]:
```suggestion
// The compiler throws an error if the function is provided a value other than a number...
```

File: lib/node_modules/@stdlib/number/float64/to-json/examples/index.js [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+var number2json = require( '../lib' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/examples/index.js]:
```suggestion
var number2json = require( './../lib' );
```

We include the leading `./`, per project conventions.

@Planeshifter This may be a good candidate for a new lint rule.

File: lib/node_modules/@stdlib/number/float64/to-json/lib/index.js [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Return a JSON representation of a `Buffer`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/index.js]:
```suggestion
* Return a JSON representation of a number.
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+* @throws {TypeError} must provide a number
+* @returns {Object} JSON representation
+*
+* @example
+* var json = number2json( NaN );
+* // returns { 'type': 'float64', 'value': 'NaN' }
+*/
+function number2json( x ) {
+	if ( typeof x !== 'number' ) {
+		throw new TypeError( format( 'invalid argument. Must provide a number. Value: `%d`.', x ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
		throw new TypeError( format( 'invalid argument. Must provide a number. Value: `%s`.', x ) );
```

I specified the wrong specifier: https://github.com/stdlib-js/stdlib/tree/3814402c0894a6379382ce657e5587738685da27/lib/node_modules/%40stdlib/string/format.

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+* var str = JSON.stringify( number2json( NaN ) );
+* // returns '{"type":"float64","value":"NaN"}'
+*/
+declare function number2json( x: number ): any;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts:31]: @Planeshifter Should we use generics here? Or actually, maybe better would be to use overloads with numeric literals. E.g.,

```
declare function number2json( x: NaN ): NANRepresentation;
declare function number2json( x: Infinity ): PosInfRepresentation;
declare function number2json( x: -Infinity ): NegInfRepresentation;
...
```

**Comment by kgryte** [comment]
@Pranavchiku You need to add the `package.json` file. This file is also needed prior to running/committing benchmarks.

**Comment by Pranavchiku** [comment]
Yeah sure, I will do that :)

**Comment by Pranavchiku** [comment]
So for ```benchmark.js```, we wont be having ```pkg+'::built-in'``` right?

**Comment by Pranavchiku** [comment]
Do we need ```benchmark.length.js``` in this?

**Comment by kgryte** [comment]
No need for `benchmark.length.js`. We use that for testing over varying array lengths. This package does not concern arrays.

File: lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+var bench = require( '@stdlib/bench' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var pkg = require( './../package.json' ).name;
+var toJson = require( './../lib' );
+
+
+// MAIN //
+
+bench( pkg, function benchmark( x ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js]: Instead of `x`, use the canonical `b` variable name. I.e., `benchmark( b ) { `.

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js]: Okay :)

File: lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+	}
+	x.toc();
+	if ( typeof v !== 'number' ) {
+		x.fail( 'should return a number' );
+	}
+	x.pass( 'benchmark finished' );
+	x.end();
+});
+
+bench( pkg+'::number,large', function benchmark( x ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js]: You can get rid of this benchmark. The magnitude of a finite number should not affect behavior and hard to imagine why it would.

File: lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+
+// MAIN //
+
+bench( pkg, function benchmark( x ) {
+	var v;
+	var i;
+
+	x.tic();
+	for ( i = 0; i < x.iterations; i++ ) {
+		v = toJson( NaN );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js]: Instead of testing multiple values like this, only test one at a time. See benchmarks which use the `values[ i%values.length ]` pattern.

File: lib/node_modules/@stdlib/number/float64/to-json/package.json [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+    "windows"
+  ],
+  "keywords": [
+    "stdlib",
+    "stdtypes",
+    "utils",
+    "util",
+    "utilities",
+    "utility",
+    "binary",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/package.json]:
```suggestion
    "number",
```

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+
+## Usage
+
+```javascript
+var number2json = require( '@stdlib/number/float64/to-json' );
+```
+
+#### number2json( x )
+
+Returns a `JSON` representation of a `number`.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]: I was unable to add hyperlink in `JSON`, ```es-lint``` was throwing some error

**Comment by Pranavchiku** [comment]
TODO: 
- [x] Add ```repl.txt``` in docs
- [x] Add hyperlink for ```JSON``` in ```README.md```

@kgryte you can have a review :)

File: lib/node_modules/@stdlib/number/float64/to-json/README.md (hunk 25 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Can you paste the error here?

File: lib/node_modules/@stdlib/number/float64/examples/index.js (hunk 12 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/examples/index.js]: @Pranavchiku Would you mind removing these changes? This is the example file for the parent namespace, not for the package.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 26]
```diff
@@ -0,0 +1,99 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# number2json
+
+> Return a `JSON` representation of a `number`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]:
```suggestion
> Return a JSON representation of a number.
```

By convention, we don't typically use backticks in the top-level description.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 27]
```diff
@@ -0,0 +1,99 @@
...
+
+## Usage
+
+```javascript
+var number2json = require( '@stdlib/number/float64/to-json' );
+```
+
+#### number2json( x )
+
+Returns a `JSON` representation of a `number`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]:
```suggestion
Returns a JSON representation of a number.
```

Our conventions have changed a bit (however, older conventions can still be found throughout the project), such as typically we only place in backticks variable or option names, not built-in data types.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+
+</section>
+
+<!-- /.references -->
+
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
+* * *
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]: Looks like this file is missing some sections. See another package README and insert the respective empty sections. You can also refer to the [README snippet](https://raw.githubusercontent.com/stdlib-js/stdlib/develop/tools/snippets/README.md).

Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: This may have also been why you were getting lint errors when attempting to insert a link.

File: lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+var bench = require( '@stdlib/bench' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var pkg = require( './../package.json' ).name;
+var toJson = require( './../lib' );
+
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js:32]: Nice benchmarks!

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts [hunk 30]
```diff
@@ -0,0 +1,38 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/// <reference types="node"/>
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/index.d.ts]:
```suggestion
```

We don't need types for Node.js. For `@stdlib/buffer/to-json` we did, as `Buffer` is a Node.js class.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md (hunk 28 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Yeah I actually removed that "See Also" section,

File: lib/node_modules/@stdlib/number/float64/to-json/docs/types/test.ts [hunk 31]
```diff
@@ -0,0 +1,41 @@
...
+
+import number2json = require( '../../lib/main' );
+
+
+// TESTS //
+
+// The function returns an object...
+{
+	const num = 3.14;
+	number2json( num ); // $ExpectType JSON
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/docs/types/test.ts:27]: This should fail linting; however, don't worry about changing now, as we need to update the TypeScript declarations to return richer information. This can be done on our side upon merge.

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js [hunk 32]
```diff
@@ -0,0 +1,69 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]:
```suggestion
* Copyright (c) 2022 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js (hunk 29 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/benchmark/benchmark.js:32]: Thank you :))

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js (hunk 32 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]: Sorry for this, I will update that :)

File: lib/node_modules/@stdlib/number/float64/to-json/test/test.js [hunk 33]
```diff
@@ -0,0 +1,130 @@
...
+var tape = require( 'tape' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var isPlainObject = require( '@stdlib/assert/is-plain-object' );
+var toJSON = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/test/test.js:32]: Nice work on the tests!

File: lib/node_modules/@stdlib/number/float64/to-json/README.md (hunk 28 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Yeah, as mentioned in the comment, you don't need to manually edit that section, as it is something that we automatically generate via tooling. For your purposes, it is enough (as in the snippet) to have an empty section.

```
<section class="related">

</section>

<!-- /.related -->
```

File: lib/node_modules/@stdlib/number/float64/to-json/lib/main.js (hunk 32 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/to-json/lib/main.js]: No worries!

@Planeshifter Maybe we can create some tooling which (a) lints the copyright year for new files and (b) auto-corrects this via a GitHub action for PRs adding new files. A combination of ESLint and ESlint fix.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md (hunk 28 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Okay sure ! Will be doing the changes :)

Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Great! Thanks!

**Comment by kgryte** [comment]
@Pranavchiku Now that you've added the `package.json` file, you should be able to add the `repl.txt` file.

File: lib/node_modules/@stdlib/number/float64/to-json/test/test.js (hunk 33 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/test/test.js:32]: Thank you :))

**Comment by Pranavchiku** [comment]
Yeah, I actually tried still it gives me some error 
I will post the logs in few minutes (by again trying it)

**Comment by kgryte** [comment]
Feel free to paste both the `repl.txt` file along with the error.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md (hunk 25 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]:
```
Linting file: lib/node_modules/@stdlib/number/float64/to-json/README.md
lib/node_modules/@stdlib/number/float64/to-json/README.md
  23:12-23:40  error  Found reference to undefined definition  no-undefined-references  remark-lint
  45:11-45:39  error  Found reference to undefined definition  no-undefined-references  remark-lint

✖ 2 errors
make: *** [/home/pranavchiku/stdlib/tools/make/lib/lint/markdown/Makefile:72: lint-markdown-files] Error 1

Markdown lint errors.
```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: ```Returns a [JSON][http://www.json.org/] representation of a number.``` 
This is how I am adding the hyperlink

Comment (kgryte) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Ah. Right. This should be `[JSON][json]`. And then in the `<section class="links">` section at the bottom, do

```
[json]: http://www.json.org/
```

In short, simply replicate what we do in [`@stdlib/buffer/to-json`](https://raw.githubusercontent.com/stdlib-js/stdlib/develop/lib/node_modules/%40stdlib/buffer/to-json/README.md).

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Yeah, I just noticed that, sorry for that

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/number/float64/to-json/README.md]: Resolved :)

**Comment by Pranavchiku** [comment]
I guess all the files are present in there, This PR can have a final review :)

Do suggest the changes and files that I missed out to add.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 34]
```diff
@@ -0,0 +1,130 @@
...
+
+## See Also
+
+<!-- -   <span class="package-name">[`@stdlib/array/to-json`][@stdlib/array/to-json]</span><span class="delimiter">: </span><span class="description">return a JSON representation of a typed array.</span>
+-   <span class="package-name">[`@stdlib/buffer/reviver`][@stdlib/buffer/reviver]</span><span class="delimiter">: </span><span class="description">revive a JSON-serialized Buffer.</span> -->
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]:
```suggestion
```

Removing this content as this content is auto-generating, as mentioned in L97.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 35]
```diff
@@ -0,0 +1,123 @@
...
+[json]: http://www.json.org/
+
+<!-- [@stdlib/buffer/ctor]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/buffer/ctor -->
+
+<!-- <related-links> -->
+
+<!-- [@stdlib/array/to-json]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/to-json -->
+
+<!-- [@stdlib/buffer/reviver]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/buffer/reviver -->
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]:
```suggestion
```

Removing unrelated content.

File: lib/node_modules/@stdlib/number/float64/to-json/README.md [hunk 36]
```diff
@@ -0,0 +1,115 @@
...
+<!-- /.related -->
+
+<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="links">
+
+[json]: http://www.json.org/
+
+<!-- </related-links> -->
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/to-json/README.md]:
```suggestion
```

**Review by kgryte — CHANGES_REQUESTED** [review]
@Pranavchiku Looks good. Only thing left is to address this comment: https://github.com/stdlib-js/stdlib/pull/527/files#r988537274.

**Comment by Pranavchiku** [comment]
Okay, for this do you want to remove folder "Examples" in "to-json" ? (as I was not clear what to do)

**Comment by kgryte** [comment]
No, no. Notice that this PR changes two different `examples/index.js` files. I am asking for the revisions to the parent namespace to be reverted.

File: lib/node_modules/@stdlib/number/float64/examples/index.js (hunk 12 above)
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/examples/index.js]:
```suggestion
```

File: lib/node_modules/@stdlib/number/float64/examples/index.js [hunk 37]
```diff
@@ -19,6 +19,16 @@
...
 console.log( objectKeys( ns ) );
+
+var str = JSON.stringify( number2json( NaN ) );
+console.log( str );
+
+str = JSON.stringify( number2json( Infinity ) );
+console.log( str );
+
+str = JSON.stringify( number2json( -Infinity ) );
+console.log( str );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/number/float64/examples/index.js]:
```suggestion
```

**Comment by kgryte** [comment]
I've specified the suggested changes.

**Review by kgryte — APPROVED** [review]
@Pranavchiku Thanks! This looks good. I'll merge now.
